feat(v8): Update node docs for v8 (#9906)
getsentry/sentry-docs: includes/node-esm-compatibility.mdx

## Using Sentry SDK with ESM

Instrumentation works out of the box for CommonJS (CJS) applications based on `require()` calls. This means that as long as your application is either natively in CJS, or compiled at build time to CJS (which is the case for most TypeScript apps, for example), everything will work without any further setup.

> Because of compatibility issues, today the Sentry SDK does not support ESM (based on `import`). We are working on a solution for this, and hope to support this soon.

Reviewer: ```suggestion
Due to compatibility issues, the Sentry SDK does not currently support ESM (based on `import`). We are working on a solution for this and hope to support this soon.
```
